### ISSUE TITLE
[BUG] Events emitted are all keyed by system name 

### ISSUE THREAD

**Issue by broody** [dojoengine/dojo#824]
**Describe the bug**
Using `emit!` macro, it seems like emitted events are keyed by the system name in hex. If we emit two different events in `execute` both use the same key making it hard to parse from frontend.

**To Reproduce**
1. Use ecs example: `sozo execute spawn` and `sozo execute move -c 0` 
2. Expect the `Moved` event to be picked up by `sozo events Moved --chunk-size 1` but it returns empty
3. Use probe to inspect receipt and find event keyed with `move`

**Expected behavior**
Events are keyed by hash of Event Name
like how world events are done

**Comment by lambda-0x** [comment]
i think i found the bug, key name is only appened to keys list when `append_keys_and_data` is called on [enum event type](https://github.com/starkware-libs/cairo/blob/98af2f86dfd19a75a8ebe7870e4f69b57fdeb756/crates/cairo-lang-starknet/src/plugin/plugin_test_data/contracts/events#L165-L178).

but we are calling `emit!` directly on the event type, ref: https://github.com/dojoengine/dojo/blob/83694ca5f71bec5a6f9e43ab2fae3dfa481cc938/examples/ecs/src/systems.cairo#L69

`emit!` expands to calling `append_keys_and_data` on the passed in type.

So we either need to require users of `emit!` to wrap it before calling or do it ourself in generated code.

**Comment by lambda-0x** [comment]
so there are two things here:
- we need to add a root level `enum Event` type
- when calling `emit!`, we need to wrap the event we want to emit in this enum type
